### PR TITLE
Permissions UI Changes

### LINKED ISSUE #2248: Role still says 'Device Owner' in profile page
sign in as a superuser and click on 'Profile'

subtask of #2201

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

Will be addressing #2201
* Added a `<permissions-icon>` core component. Uses keys. Made use of it in the permission and profile pages.
* No icon for no permissions.
* 'Some' permissions -> 'Limited' permissions Note that screenshots are slightly outdated
* Made icon it's own column.
* Display permissions in profile page.
* Adds role to app bar menu. 
* Username is now visible at all times. Addresses #2260 See 2nd to last screenshot.
* Please review carefully
* Fixes #2248
* Addresses #2115 - Dropdown is currently disabled. Looks bad because the dropdown has not been implemented to style guide spec. Can just remove it instead? See last screenshot 

127 0 0 1-8000-management-device

127 0 0 1-8000-management-device 1

127 0 0 1-8000-user

127 0 0 1-8000-user 3

127 0 0 1-8000-user 2

127 0 0 1-8000-management-facility

127 0 0 1-8000-management-facility 1

### CODE REVIEW

File: kolibri/auth/api.py [hunk 1]
```diff
@@ -233,21 +233,24 @@ def get_session(self, request):
                    'user_id': user.id,
                    'can_manage_content': user.can_manage_content}
         roles = Role.objects.filter(user_id=user.id)
-        if len(roles) is not 0 or user.is_superuser:
+
+
+        if len(roles) is 0:
```
Comment (christianmemije) [inline kolibri/auth/api.py]: Ideally we should not be mixing superuser (permission type) with role type. They should be separate. But for now the change I made is that superuser learners also have the learner role.

File: kolibri/core/assets/src/state/getters.js [hunk 2]
```diff
@@ -1,29 +1,50 @@
...
+  return state.core.session.kind.includes(UserKinds.SUPERUSER);
+}
+function getUserPermissions(state) {
+  const permissions = {};
+  permissions.can_manage_content = state.core.session.can_manage_content;
+  return permissions;
+}
+
+function userHasPermissions(state) {
+  return some(getUserPermissions(state), permission => permission === true);
```
Comment (jonboiser) [inline kolibri/core/assets/src/state/getters.js]: `some(getUserPermissions(state), /* no predicate */)` should be enough. If the value `permissions` is Boolean, then the predicate for `some` is basically the identity, which is the default.

**Comment by jonboiser** [comment]
If you change your Role on 'Manage Users' page, it does not get automatically updated in the App Bar. You might need to update the edit user action add an appropriate mutation.

Another thing that is odd in this new permissions structure is that if you switch back from Admin to Learner, you will be forced to the Learn page (because learners would never be able to see Manage Users page), even though  you still have superuser permissions.

** Right after changing from Learner to Admin**
screen shot 2017-09-28 at 4 52 15 pm

File: kolibri/plugins/management/assets/src/views/index.vue [hunk 3]
```diff
@@ -9,7 +9,7 @@
       <component class="manage-content page" :is="currentPage"/>
     </div>
 
-    <auth-message v-else authorizedRole="admin" />
+    <auth-message v-else :details="$tr('admninOrSuperuser')" />
```
Comment (jonboiser) [inline kolibri/plugins/management/assets/src/views/index.vue]: Typo here? `adminin` == `admin`. Check to see if it matches `options.$trs` too.

Comment (christianmemije) [reply kolibri/plugins/management/assets/src/views/index.vue]: > Check to see if it matches `options.$trs` too.

Can you clarify?

**Comment by christianmemije** [comment]
> Another thing that is odd in this new permissions structure is that if you switch back from Admin to Learner, you will be forced to the Learn page (because learners would never be able to see Manage Users page), even though you still have superuser permissions.

Nice catch. Will fix by removing this redirect. 

https://github.com/christianmemije/kolibri/blob/33c7be7cca14823107913f1b4acb4bbc19fd2ef1/kolibri/plugins/management/assets/src/state/actions.js#L469

**Comment by christianmemije** [comment]
@jonboiser Addressed comments.

**Pull request merged**